test-cypress/integration/repository/repositories.spec.js

```diff
@@ -0,0 +1,404 @@
+    });
+
+    it('should allow to edit existing repository', () => {
+        const newTitle = 'Title edit';
+        const newBaseUrl = 'http://example.org/wine#';
+
+        createRepository();
+        typeRepositoryId(repositoryId);
+        typeRepositoryTitle('Title');
+        saveRepository();
+
+        editRepository(repositoryId);
+
+        // Some fields should be disabled
+        getRepositoryRulesetMenu().should('be.disabled');
+        getRepositoryDisableSameAsCheckbox().should('be.disabled');
+
+        typeRepositoryTitle(newTitle);
+        typeRepositoryBaseURL(newBaseUrl);
+        getRepositoryContextIndexCheckbox().check();
+
+        getSaveRepositoryButton().click();
+        confirmModal();
+        waitLoader();
+
+        // See the title is rendered in the repositories list
+        getRepositoryFromList(repositoryId).should('contain', newTitle);
+
+        // Go to edit again to verify changes
+        editRepository(repositoryId);
+
+        getRepositoryTitleField().should('have.value', newTitle);
+        getRepositoryBaseURLField().should('have.value', newBaseUrl);
+        getRepositoryContextIndexCheckbox().should('have.value', 'true');
+    });
+
+    it('should allow to delete existing repository', () => {
+        createRepository();
+        typeRepositoryId(repositoryId);
+        saveRepository();
+
+        selectRepoFromDropdown(repositoryId);
+
+        getRepositoryFromList(repositoryId)
+            .find('.repository-actions .delete-repository')
+            // Forcefully clicking it due to https://github.com/cypress-io/cypress/issues/695
+            .should('be.visible')
+            .and('not.be.disabled')
+            .click({force: true});
+
+        confirmModal();
+
+        getRepositoriesList().should('not.contain', repositoryId);
+
+        // Check the repo has been deselected and is not present in the repo dropdown menu
+        getRepositoriesDropdown().click().within(() => {
+            cy.get('.no-selected-repository').should('be.visible');
+            cy.get('.dropdown-menu .dropdown-item').should('not.contain', repositoryId);
+        });
+    });
+
+    const REPO_LIST_ID = '#wb-repositories-repositoryInGetRepositories';
+
+    function getRepositoriesList() {
+        return cy.get(REPO_LIST_ID);
+    }
+
+    function getLocationsList() {
+        return cy.get('.locations-table tr');
+    }
+
+    function getRepositoryFromList(repository) {
+        return getRepositoriesList()
+            .find('.repository')
+            .contains(repository)
+            .parentsUntil(REPO_LIST_ID + ' tbody')
+            .last();
+    }
+
+    function editRepository(repositoryId) {
+        getRepositoryFromList(repositoryId)
+            .should('be.visible')
+            .find('.repository-actions .edit-repository')
+            // Forcefully clicking it due to https://github.com/cypress-io/cypress/issues/695
+            .should('be.visible')
+            .and('not.be.disabled')
```

Comment:
Maybe the disabled check is redundant here as the click would blow in case it's not enabled

Reply:
The click is forced so I doubt it would blow.